### ISSUE TITLE
Fix Request: AvoidImplicitlyRecompilingRegex - always flag String.replaceAll()

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#115]
Currently String.replaceAll() is only flagged when the regexp is larger than some small size. 

The String.replaceAll() method does not have a fast-path implementation as far as I checked, so better always avoid the Pattern compilation and reuse a precompiled Pattern, regardless of regexp size.

Example:

```java
private String extractBody(final HttpServletRequest request) throws IOException {    
     String body = request.getReader().lines().collect(Collectors.joining(System.lineSeparator())).replaceAll("\"", ""); // not flagged
     if (log.isDebugEnabled()) {
        log.debug("Body: {}", body.replaceAll("\\s+","")); // flagged   
    }
```

**Comment by stokpop** [comment]
Todo: check for other without fast-path: only flag always when there is no fast-path

**Comment by stokpop** [comment]
For just one character `replaceAll("a","b")` the improvement seen with JMH for reuse of pattern is ~1.4x
Is this worth replacing always? Maybe create a rule for "info" level for these cases, so developer can decide what to do.

```
Benchmark                                        Mode  Cnt        Score        Error  Units
LetsRegex.stringReplaceAllSingleChar            thrpt    4  3025408,961 ± 151267,703  ops/s
LetsRegex.stringReplaceAllSingleCharOnePattern  thrpt    4  4288050,487 ±  36875,596  ops/s
```

For one character with some regexp logic, e.g. `replaceAll("(a)+","b")` the improvement seen with JMH for reuse of pattern is ~2.5x

```
Benchmark                                               Mode  Cnt        Score       Error  Units
LetsRegex.stringReplaceAllSingleCharComplexPattern     thrpt    4  1523619,882 ± 19331,396  ops/s
LetsRegex.stringReplaceAllSingleCharOneComplexPattern  thrpt    4  3787124,686 ± 42448,893  ops/s
```

See benchmark code: https://github.com/stokpop/performance-playground/blob/main/src/main/java/nl/stokpop/jmh/LetsRegex.java

**Comment by jborgers** [comment]
The speedup is not much, at the cost of readability. We leave it as is.
